s2wav/Dimension helper functions.py

```diff
@@ -0,0 +1,373 @@
+        j (int): Wavelet scale.
+    
+    Returns:
+        el_min (int).
+    '''
+    if kernel.lower() == "s2dw":
+        return math.ceil(lam ** (j-1))
+    else:
+        #This code currently does not support other wavelet types
+        return print("Invalid wavelet type.")
+
+
+
+def j_max(L: int, lam: float) -> int:
+    '''Computes needlet maximum level required to ensure exact reconstruction.
+
+    Args:
+        L (int): Upper harmonic band-limit.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+    
+    Returns:
+        j_max (int).
+    '''
+    return math.ceil(log(L) / log(lam))
+
+
+
+
+
+#files originally from s2let_helper
+
+def n_phi(sampling_scheme: str, L: int) -> int:
+    '''Returns the number of phi samples.
+
+    Args:
+        sampling_scheme (str): Either S2LET_SAMPLING_MW or S2LET_SAMPLING_MW_SS.
+        L (int): Upper harmonic band-limit.
+
+    Returns:
+        nphi (int): Number of phi samples.
+    '''
+    if sampling_scheme.lower() == "mwss":
+        return sampling_mw_ss_nphi(L)
+    elif sampling_scheme.lower() == "mw":
+        return ampling_mw_nphi(L)
+
+
+def n_theta(sampling_scheme: str, L: int) -> int:
+    '''Returns the number of theta samples.
+
+    Args:
+        sampling_scheme (str): Either S2LET_SAMPLING_MW or S2LET_SAMPLING_MW_SS.
+        L (int): Upper harmonic band-limit.
+
+    Returns: 
+        ntheta (int): Number of theta samples.
+    '''
+    if sampling_scheme.lower() == "mwss":
+        return sampling_mw_ss_ntheta(L)
+    elif sampling_scheme.lower() == "mw":
+        return sampling_mw_ntheta(L)
+
+
+def n_px(sampling_scheme: str, L: int) -> int:
+    '''Returns the number of phi samples multiplied by the number of theta samples.
+    
+    Args:
+        sampling_scheme (str): Either S2LET_SAMPLING_MW or S2LET_SAMPLING_MW_SS.
+        L (int): Upper harmonic band-limit.
+    
+    Returns:
+        (int): nphi * ntheta.
+    '''
+    return n_phi(sampling_scheme, L) * n_theta(sampling_scheme, L)
+
+
+def n_lm(L: int) -> int:
+    '''Returns the square of the harmonic band-limit.
+    
+    Args:
+        L (int): Upper harmonic band-limit.
+
+    Returns:
+        Square of L (int).
+    '''
+    return L*L
+
+
+def n_lm_scal(upsample: bool, L: int, kernel: str, J_min: int, lam: float) -> int:
+    '''Computes the square of the band-limit, after determining what the value of the band-limit is.
+
+    Args:
+        upsample (bool): Boolean parameter which determines whether to store the scales at j_max resolution or its own resolution.
+        L (int): Upper harmonic band-limit.
+        kernel (str): The wavelet type.
+        J_min (int): First wavelet scale to be used.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+    
+    Returns:
+        The square of the band-limit (int).
+    '''
+    if bandlimit == upsample:
```

Comment:
This doesn't make sense, it should read:

`if upsample:`

I.e. if upsample is true then...

Reply:
Similarly, does the code on [this line (232)](https://github.com/astro-informatics/s2wav/pull/25/files#diff-3ad13bfa2e52184a548715a839b096d4943bc2e771015ea813fd9e588e059069R231) also not make sense? For reference, the associated line in the C-code is https://github.com/astro-informatics/s2let/blob/main/src/main/c/s2let_helper.c#L48

Reply:
Should I put `if upsample == false` instead?

Reply:
I think line 232 is fine. Basically upsample should be a boolean which is used to toggle between two different settings. So line 232 is already doing 
`if upsample == false:`
because that is just a slightly less concise way of doing
`if not upsample:`
if that makes sense?